test/unit/test_providers_kms_master_key_provider.py

```diff
@@ -124,9 +129,11 @@ def test_client_no_region_name_with_default(self, mock_add_client):
         test._regional_clients[sentinel.default_region] = sentinel.default_client
         client = test._client("")
         assert client is sentinel.default_client
-        mock_add_client.assert_called_once_with(sentinel.default_region)
+        mock_add_client.assert_called_with(sentinel.default_region)
 
     def test_client_no_region_name_without_default(self):
+        if default_region_is_set():
```

Comment:
The intent of this test is to verify what happens if a client is created that cannot determine the default region, and then we attempt to obtain a client using a key ID that is not a complete Arn.

Some better options for fixing this test would be one of:
* use the unknown-region botocore session mentioned earlier
* manually set the instance's default region to `None` before calling `test._client`
* two parts:
    1. change this test to instead mock out `_region_from_key_id` and then simply verify that it is called with the expected values
    1. add stand-alone tests for the `_region_from_key_id` function

Reply:
Fixed this by creating a session with no default region.